### ISSUE TITLE
Openvino Usage

### ISSUE THREAD

**Issue by I3orn2FLY** [ultralytics/yolov5#1193]
## Additional context
I converted yolov5 to ONNX using export.py, then I cloned the OpenVINO toolkit and installed an inference engine. Then I got the IR files by using mo.py in the cloned OpenVINO toolkit. I managed to run the IR model somehow by using sample codes in OpenVINO. However, now I do not know how to decode the output of the model. Is the procedure the same as in samples of yolov3? Does someone have a script in c++? If yes, I would be more than happy for clues.

**Comment by I3orn2FLY** [comment]
Hi guys, I managed to decode the output of the model. First of all, look at the yolo.py, Detect class where the forward function is written, it seems like when the model is converted to ONNX anything which is covered in if statements are not exported, therefore, you will need to rewrite that part of the code in your project as well. Here is the if statement, which you should implement in your project:https://github.com/ultralytics/yolov5/blob/bd5cfff8fbdcc079ff901cb7c99fab916be06f57/models/yolo.py#L47. Note that you can rewrite Detect.py so that at least sigmoid is run before this if statement, so it will be exported with sigmoid. There are also some details about yolo decoding (regarding anchor boxes / grid) which you should know if you want to rewrite it in your project.

**Comment by ALMerrill** [comment]
@I3orn2FLY, do you have a good understanding of what the code is doing inside of that if statement? I'm working on doing this for coreml, but I'm having a hard time understanding what the lines inside the if statement are doing in order to convert them to coreml. Do you have any insight on this?

**Comment by ZhixiongSun** [comment]
Hello, you can get an executable cpp code here: https://github.com/fb029ed/yolov5_cpp_openvino,
this is based on yolov5 v3.1

**Comment by glenn-jocher** [comment]
@I3orn2FLY @ALMerrill @ZhixiongSun good news 😃! Your original issue may now be fixed ✅ in PR #6057. This PR adds native YOLOv5 OpenVINO export:

```bash
python export.py --weights yolov5s.pt --include openvino  # export to OpenVINO
```
![image](https://user-images.githubusercontent.com/26833433/147145516-b4f09a44-5679-472a-b512-97c51adffefc.png)


To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
